apps/judicial-system/web/src/graphql/gql.ts

```diff
@@ -0,0 +1,353 @@
+      state
+      policeCaseNumber
+      accusedNationalId
+      accusedName
+      accusedAddress
+      court
+      arrestDate
+      requestedCourtDate
+      requestedCustodyEndDate
+      lawsBroken
+      custodyProvisions
+      requestedCustodyRestrictions
+      caseFacts
+      witnessAccounts
+      investigationProgress
+      legalArguments
+      comments
+      # prosecutorId
+      prosecutor {
+        nationalId
+        name
+        title
+        mobileNumber
+        email
+        role
+      }
+      courtCaseNumber
+      courtStartTime
+      courtEndTime
+      courtAttendees
+      policeDemands
+      accusedPlea
+      litigationPresentations
+      ruling
+      rejecting
+      custodyEndDate
+      custodyRestrictions
+      accusedAppealDecision
+      accusedAppealAnnouncement
+      prosecutorAppealDecision
+      prosecutorAppealAnnouncement
+      # judgeId
+      judge {
+        nationalId
+        name
+        title
+        mobileNumber
+        email
+        role
+      }
+      # notifications
+    }
+  }
+`
+
+export const TransitionCaseMutation = gql`
+  mutation TransitionCaseMutation(
+    $id: ID!
+    $transitionCase: TransitionCaseDto!
+  ) {
+    transitionCase(id: $id, transitionCase: $transitionCase) {
+      id
+      created
+      modified
+      state
+      policeCaseNumber
+      accusedNationalId
+      accusedName
+      accusedAddress
+      court
+      arrestDate
+      requestedCourtDate
+      requestedCustodyEndDate
+      lawsBroken
+      custodyProvisions
+      requestedCustodyRestrictions
+      caseFacts
+      witnessAccounts
+      investigationProgress
+      legalArguments
+      comments
+      # prosecutorId
+      prosecutor {
+        nationalId
+        name
+        title
+        mobileNumber
+        email
+        role
+      }
+      courtCaseNumber
+      courtStartTime
+      courtEndTime
+      courtAttendees
+      policeDemands
+      accusedPlea
+      litigationPresentations
+      ruling
+      rejecting
+      custodyEndDate
+      custodyRestrictions
+      accusedAppealDecision
+      accusedAppealAnnouncement
+      prosecutorAppealDecision
+      prosecutorAppealAnnouncement
+      # judgeId
+      judge {
+        nationalId
+        name
+        title
+        mobileNumber
+        email
+        role
+      }
+      # notifications
+    }
+  }
+`
+
+export const SendNotificationMutation = gql`
+  mutation SendNotificationMutation($id: ID!) {
+    sendNotification(id: $id) {
+      id
+      created
+      caseId
+      type
+      message
+    }
+  }
+`
+
+export const RequestSignatureMutation = gql`
+  mutation RequestSignatureMutation($id: ID!) {
+    requestSignature(id: $id) {
+      controlCode
+      documentToken
+    }
+  }
+`
+
+export const ConfirmSignatureQuery = gql`
+  query ConfirmSignatureQuery($id: ID!, $documentToken: String!) {
+    confirmSignature(id: $id, documentToken: $documentToken) {
+      documentSigned
```

Comment:
If an ID is requested as well, then Apollo will know how to update its store.

Reply:
I do not understand this comment. Here we are querying a document signature service, asking for the result of a previously requested document signature. The signature is related to a court case that is stored in a database and has an ID, but the the court case itself is not returned from this query. The returned object is transient, not stored and only relevant while showing the user whether the document signing succeeded or not.

Reply:
My confusion was due to the name of the parameter `id`.
Is this `id` the id of the confirm-signature which is being queried, or the id of another object, e.g. document-signature, which would lead to a more descriptive query:
```graphql
confirmSignature(documentSignatureId: $id, ...) {
  ...
}
```

Reply:
Got you, I will clarify the naming. I should not use ID here I guess as I do not want Apollo to update anything.